### ISSUE TITLE
How to use HLS encryption ?

### ISSUE THREAD

**Issue by mrfrieza** [arut/nginx-rtmp-module#1256]
Hello,

I don't think I'm bad with nginx conf files, but I'm stucked...

I can't get HLS encryption to work and I would like to know if someone could share their bit of conf file so that I can learn from it ?

I'm not using https.. I don't know if it's possible without it.. I'm creating m3u8 by streaming with obs through rtmp. Then I run tests to watch it with 4G / Other wifi to see if it's playable (opened ports, works just fine). 

Though, I'd like it to be more secured.. Thanks !

**Comment by dromer** [comment]
Just serve the hls fragments over ssl?
Why would you not use https? and how would you expect hls itself to be encrypted?

**Comment by misiek08** [comment]
@dromer he's talking about fragment encryption. It's a thing, because some producers require that even temporary downloaded files on users computers should be encrypted.

I'm on mobile and Apple docs sucks on small devices, but probably this are the docs: https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/HLS_Sample_Encryption/Intro/Intro.html

OP: show the config you already have, so we can help with your exact case.

**Comment by dromer** [comment]
I see @misiek08 

I'd still use ssl for serving the fragments to start with ;)

**Comment by misiek08** [comment]
Yes, SSL nowadays is must have, but for some clients it's not enough. Waiting for current OP's config to see what's wrong.
